### DISCUSSION TITLE
s2spy

### DISCUSSION THREAD

**Discussion by Akshayksagar** [AI4S2S/s2spy#170] [General]
hey, iam unable to install s2spy package and also s2spy.time,rgdr etc . Please help

**Comment by geek-yang** [comment]
Hi, thanks for asking. Could you please provide some information about your setup? For instance, your operating system info, your python version, and the error message?

**Comment by Akshayksagar** [comment]
I am using i5 9th gen, 16 gb ram, windows, python 3.9 and if i am doing pip
install s2spy its executed but when i am importing in python. It is saying
that no module name s2spy and s2spy.time/s2spy.rgdr.

**Comment by Akshayksagar** [comment]
![Screenshot (192)](https://github.com/AI4S2S/s2spy/assets/46590370/9e1930a9-1254-4868-a444-62ee05ad90e9)
this is the error i am getting

**Reply by geek-yang** [reply]
@Akshayksagar Thanks a lot for your description. In the latest release of `s2spy`, `time` module has been deprecated and it becomes a spin-off package of `s2spy`, called [`lilio`](https://github.com/AI4S2S/lilio). To create calendar and manage your data, you need `lilio`. Here is the documentation about how to use it https://lilio.readthedocs.io/en/latest/.

`s2spy` can help you with simple preprocessing and dimensionality reduction (e.g. RGDR). And here is the documentation of how to use s2spy https://ai4s2s.readthedocs.io/en/latest/.

I think to build your workflow, you need both of them. 😉

**Reply by Akshayksagar** [reply]
![Screenshot 2023-07-07 232151](https://github.com/AI4S2S/s2spy/assets/46590370/353104f8-89c7-4672-a570-fec191834453)
i am getting this error while getting cluster regions, any views ?

**Reply by geek-yang** [reply]
I guess it is mainly due to your input data, they must be `xarray.Dataarray` or `xarray.Dataset`. Could you please check your input data? Just check the example  https://github.com/AI4S2S/s2spy/blob/main/docs/notebooks/tutorial_RGDR.ipynb.

I will soon update this notebook to make test data accessible, and therefore make sure that the user can run the notebook themselves. Please keep an eye.

**Reply by geek-yang** [reply]
We just update the test data used in the tutorial notebook. You should be able to access the data via the link or in this repo https://github.com/AI4S2S/cookbook/tree/main/data. More details about the data can also be found there. Let me know if you have more questions.
